src/screens/CurrencySelectModal.js

```diff
@@ -186,13 +186,22 @@ export default function CurrencySelectModal() {
     [handleFavoriteAsset, handleSelectAsset, type]
   );
 
-  const handleApplyFavoritesQueue = useCallback(
-    () =>
-      Object.keys(assetsToFavoriteQueue).map(assetToFavorite =>
-        updateFavorites(assetToFavorite, assetsToFavoriteQueue[assetToFavorite])
-      ),
-    [assetsToFavoriteQueue, updateFavorites]
-  );
+  const handleApplyFavoritesQueue = useCallback(async () => {
+    const addresses = Object.keys(assetsToFavoriteQueue);
+    const [assetsToAdd, assetsToRemove] = addresses.reduce(
+      ([add, remove], current) => {
+        if (assetsToFavoriteQueue[current]) {
+          add.push(current);
+        } else {
+          remove.push(current);
+        }
+        return [add, remove];
+      },
+      [[], []]
+    );
+    await updateFavorites(assetsToAdd, true);
+    await updateFavorites(assetsToRemove, false);
+  }, [assetsToFavoriteQueue, updateFavorites]);
```

Comment:
I'd move this into a then syntax so it's not blocking. Something around the lines of

`updateFavorites(assetsToAdd, true).then ( () => updateFavorites(assetsToRemove, false));`

Reply:
Alright, done. Did some testing lg2m.